imports/client/components/HuntListPage.tsx

```diff
@@ -106,26 +107,26 @@ const DiscordSelector = (props: DiscordSelectorProps) => {
   }
 };
 
-const DiscordChannelSelector = (params: DiscordSelectorParams) => {
+const DiscordChannelSelector = (params: DiscordSelectorParams & { guildId?: string }) => {
   const { ready, options } = useTracker(() => {
     const handle = Meteor.subscribe('discord.cache', { type: 'channel' });
-    const discordChannels: SavedDiscordObjectType[] = DiscordCache.find(
+    const discordChannels: SavedDiscordObjectType[] = DiscordCache.find({
+      type: 'channel',
+      'object.guild': params.guildId,
```

Comment:
Talk me through what you expect to happen if `guildId` is `undefined` here.  Does it omit that field from the filter, resulting in an enumeration of *all* text channels that the bot has access to?  Or does it only match values where `object.guild` is `undefined`?

(I guess while we're here, what do you want the behavior here to be if no guild has been selected?  I feel like guild should maybe be required if we're going to attempt to make this more scoped)

Reply:
Hmm. I guess I didn't quite think through this deeply enough. In practice, the current behavior is that `undefined` acts like `null` - it matches an object where that field is explicitly set to null or unset, which in turn matches nothing (since all channel and role objects have a `guild` property):

```
Models.DiscordCache.find().count()
36
Models.DiscordCache.find({'object.guild': null}).count()
0
Models.DiscordCache.find({'object.guild': undefined}).count()
0
```

Honestly, though, I think the behavior I actually want is for that dropdown to be disabled when `guildId` is undefined, and then it mostly doesn't matter. I'll update accordingly.

Reply:
If this component isn't going to be useful for anything if `guildId` isn't specified, then we should probably just make the param required?  Unless you're hoping to keep the field usable for *unsetting* the channel if it was previously set and then the guild becomes unconfigured, but then we shouldn't disable the field?